### ISSUE TITLE
GC Events are broken on LTTng

### ISSUE THREAD

**Issue by sywhang** [dotnet/coreclr#22306]
Using LTTng to collect traces no longer collects GC events. This is broken because the event keywords are not set on the GC side's cache of the event keywords when we enable events via LTTng/PerfCollect.

**Comment by sywhang** [comment]
Update: This is broken because GC does not know whether tracing via LTTng is enabled. The way we check whether LTTng is enabled is via the call to ```tracepoint_enabled``` which is only done from the runtime side. GC checks whether events are enabled before crossing the GC/runtime boundary via GCEventEnabled* methods, which checks only its own cache of enabled keyword/level, which won't know whether events are enabled via LTTng. 

One way to fix this would be to call helper methods that wrap ```tracepoint_enabled```, specifically the ```EventXplatEnabled*``` methods inside the ```GCEventEnabled```* methods.

**Comment by sywhang** [comment]
cc @noahfalk @maoni0 @andy-ms

**Comment by gleocadie** [comment]
@sywhang I was under the impression that keyword and level checks were left aside when FEATURE_PAR was set. Did I miss something?

https://github.com/dotnet/coreclr/blob/master/src/inc/eventtracebase.h#L104

**Comment by sywhang** [comment]
@gleocadie Do you mean `FEATURE_PAL`?  The line you're referring to is ETW stuff which has nothing to do with LTTng. There is a discussion in the PR I sent out above about on what the actual issue is and how to fix it / work around it in the meantime.

**Comment by noahfalk** [comment]
FYI a bunch of those 'ETW' named functions and macros have very misleading names at this point. Originally they meant ETW only, but now they mean 'whatever tracing system(s)' are in use on this platform and I think it would be very beneficial to rename them more accurately at some point. Unless they've managed to confuse me once again, I think the line @geocadie referenced really does relate to LTTng when run on Linux, although I am unaware that it is used in the specific case of the GC events.

**Comment by gleocadie** [comment]
@sywhang The name is really misleading and as @noahfalk pointed out, it's for 'whatever tracing system(s)'.
I will check your PR.
At Criteo, we made a fix in our fork (CoreCLR), which reads an environment variable which contains the configuration of the Provider/Keywords/Level we want. Once parsed, it works like ETW and we can filter events/keywords to be built and emitted.
The reason we did not push upstream was this https://github.com/dotnet/coreclr/issues/14327
But it's something worth, I would be ok to make a PR.
